src/generate/peripheral.rs

```diff
@@ -194,14 +195,15 @@ pub fn render(
 
     // Push all register related information into the peripheral module
     for reg in registers {
-        mod_items.extend(register::render(
-            reg,
-            registers,
-            p,
-            all_peripherals,
-            &defaults,
-            config,
-        )?);
+        mod_items.extend(
+            match register::render(reg, registers, p, all_peripherals, &defaults, config) {
+                Ok(rendered_reg) => rendered_reg,
+                Err(e) => {
+                    let res: Result<TokenStream> = Err(e);
+                    return handle_reg_error("Error redering regular register", *reg, res);
```

Comment:
typo, should also not say `regular`
```suggestion
                    return handle_reg_error("Error rendering register", *reg, res);
```